pos_tare_generate_barcode/static/src/js/screens.js

```diff
@@ -0,0 +1,200 @@
+    // the barcode label printing screen defined below.
+    var TareScreenButton = screens.ActionButtonWidget.extend({
+        template: 'TareScreenButton',
+
+        button_click: function () {
+            this.gui.show_screen('tare');
+        },
+    });
+
+    screens.define_action_button({
+        'name': 'tareScreenButton',
+        'widget': TareScreenButton,
+    });
+
+    // This is a new screen that reads weight from the electronic scale and
+    // create a barcode label encoding the weight. The screen shows a preview
+    // of the label. The user is expected to check if the preview matches what's
+    // measured on the scale. The barcode image is generated by the report
+    // module.
+    var TareScreenWidget = screens.ScreenWidget.extend({
+        template: 'TareScreenWidget',
+        next_screen: 'products',
+        previous_screen: 'products',
+        default_tare_value: 0.0,
+        weight_barcode_prefix: null,
+
+        show: function () {
+            this._super();
+            // Fetch the barcode prefix from POS barcode parser rules.
+            this.weight_barcode_prefix = this.get_barcode_prefix();
+            // Setup the proxy
+            var queue = this.pos.proxy_queue;
+            // The pooling of the scale starts here.
+            var self = this;
+            queue.schedule(function () {
+                return self.pos.proxy.scale_read().then(function (weight) {
+                    try {
+                        self.set_tare_weight(weight);
+                    } catch (error) {
+                        var title = _t("Failed to read weight from scale.");
+                        var popup = {title: title, body: error.message};
+                        self.gui.show_popup('error', popup);
+                        self.pos.proxy_queue.clear();
+                    }
+                });
+            }, {duration:150, repeat: true});
+            // Shows a barcode whose weight might be zero, but this is preferred
+            // for UI/UX reasons.
+            this.render_receipt();
+            this.lock_screen(true);
+        },
+        get_barcode_prefix: function () {
+            var barcode_pattern = this.get_barcode_pattern();
+            return barcode_pattern.substr(0, 2);
+        },
+        get_barcode_pattern: function () {
+            var rules = this.get_barcode_rules();
+            var rule = rules.filter(
+                function (r) {
+                    // We select the first (smallest sequence ID) barcode rule
+                    // with the expected type.
+                    return r.type === "tare";
+                })[0];
+            return rule.pattern;
+        },
+        get_barcode_rules: function () {
+            return this.pos.barcode_reader.barcode_parser.nomenclature.rules;
+        },
+        set_tare_weight: function (scale_measure) {
+            var weight = scale_measure.weight;
+            var measure_unit = this.pos.units.filter(
```

Comment:
@houssine78 We pick up the scale UOM from there. A scale using the Imperial scale would not be a problem there as long as the UOM exists.